km/km_hcalls.c

```diff
@@ -1432,6 +1432,23 @@ static km_hc_ret_t clone_hcall(void* vcpu, int hc, km_hc_args_t* arg)
    return HC_CONTINUE;
 }
 
+static km_hc_ret_t clone3_hcall(void* vcpu, int hc, km_hc_args_t* arg)
+{
+   // long syscall(SYS_clone3, struct clone_args *cl_args, size_t size);
+   struct clone_args* cl_args = km_gva_to_kma(arg->arg1);
+
+   if (cl_args == NULL) {
```

Comment:
we generally do end of structure check to be in bounds.

Reply:
Not sure this is actually correct. I see a lot of places where we only check the address. Can you point at examples?

Reply:
uname hcall.